cli/command/image/build.go

```diff
@@ -247,6 +253,66 @@ func runBuild(dockerCli *command.DockerCli, options buildOptions) error {
 		buildCtx = replaceDockerfileTarWrapper(ctx, buildCtx, relDockerfile, translator, &resolvedTags)
 	}
 
+	// write ctx tar to disk, to make it readable for Image Reference extraction
+	originCtx := buildCtx
+	defer originCtx.Close()
+	readbackTar, err := ioutil.TempFile("", "docker-ctx")
```

Comment:
Yes, it would be even better to build on the generic function in #31236. Ideally we would end up with a versatile wrapper that can do several things, instead of needing to stack wrappers. But stacking wrappers would still be better than writing the tar to disk temporarily.